src/bson_mapper/utility.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iostream>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+
+#include <bson_mapper/archiver_mock.hpp>
+#include <bson_mapper/bson_streambuf.hpp>
+
+/**
+ * Converts a serializable object into a BSON document value
+ * TODO This is not very clean and kind of inefficient. Maybe we should be passing references
+ * into
+ * the bson_ostream callback?
+ * @tparam T   A type that is serializable to BSON using a BSONArchiver.
+ * @param  obj A serializable object
+ * @return     A BSON document value representing the given object.
+ */
+template <class T>
+bsoncxx::document::value to_document(const T& obj) {
```

Comment:
Let's put all of these free functions in the bson_mapper namespace.
Also, I would prefer a more descriptive name. Maybe `mapping_functions.hpp`?